sympy/core/power.py

```diff
@@ -1062,15 +1062,22 @@ def _eval_is_rational(self):
                 return e.is_zero
 
     def _eval_is_algebraic(self):
-        if self.base.is_zero or (self.base - 1).is_zero:
+        def _is_one(expr):
+            try:
+                return (expr - 1).is_zero
+            except ValueError:
+                # when the operation is not allowed
+                return False
+
+        if self.base.is_zero or _is_one(self.base):
             return True
         elif self.exp.is_rational:
             if self.base.is_algebraic is False:
                 return self.exp.is_nonzero
```

Comment:
I believe this is unrelated to the above fix (let me know if I am wrong). If so, we should fix this in a PR to master.